torch/nested/_internal/ops.py

```diff
@@ -897,16 +894,38 @@ def sum_dim_IntList(func, *args, **kwargs):
                     "sum(): not supported along a ragged and non-batch dimension for NestedTensor"
                 )
             # reduction cases: (ragged)
+            # shift reduction dimension of values backward to outer dimension
+            values_transposed_backward = inp._values.permute(
+                inp._ragged_idx - 1,
+                *range(0, inp._ragged_idx - 1),
+                *range(inp._ragged_idx, inp.dim() - 1),
+            )
+
+            padded = torch.ops.aten._jagged_to_padded_dense_forward(
+                values_transposed_backward.reshape(
+                    values_transposed_backward.shape[0], -1
+                ),  # _jagged_to_padded_dense_forward requires values to be a 2D tensor
+                [inp._offsets],
+                max_lengths=[inp._max_seqlen],
+            )
+
+            # expand non-batch dimensions of padded tensor
+            padded_transposed_forward = padded.view(
+                padded.shape[0],
+                inp._max_seqlen,
+                *values_transposed_backward.shape[1:],
+            )
+
+            # shift reduction dimension of padded tensor forward to original ragged dimension
+            padded_transposed_forward = padded_transposed_forward.permute(
```

Comment:
cc @jbschlosser: I think the permute could potentially change the striding in a non-obvious way.

I don't think this is a problem (and it's what I'd expect), but wanted to flag it in case it's a concern